dht.go

```diff
@@ -507,3 +578,46 @@ func (dht *IpfsDHT) newContextWithLocalTags(ctx context.Context, extraTags ...ta
 	) // ignoring error as it is unrelated to the actual function of this code.
 	return ctx
 }
+
+func (dht *IpfsDHT) handleProtocolChanges(ctx context.Context) {
+	// register for event bus protocol ID changes
+	sub, err := dht.host.EventBus().Subscribe(new(event.EvtPeerProtocolsUpdated))
+	if err != nil {
+		panic(err)
+	}
+	defer sub.Close()
+
+	pmap := make(map[protocol.ID]bool)
```

Comment:
nit: preallocate

Reply:
this is a map, no preallocation required. Either way I think we're scrapping this map since we don't need to do protocol checks here.

Reply:
SGTM. Note, we can totally pre-allocate maps.